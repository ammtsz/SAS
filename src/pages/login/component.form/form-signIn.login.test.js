import React from "react";
import SignIn from "./form-signIn.login.component";

import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import {
  actionSetPersistence,
  actionEmailSignIn,
} from "../../../redux/user/user.actions";

describe("Component login-page <SignIn />", () => {
  const mockStore = configureMockStore();
  const mockPersistence = false;
  const store = mockStore({
    user: {
      datas: null,
      persistence: mockPersistence,
      error: null,
    },
  });
  store.dispatch = jest.fn();

  const mockNewAccount = jest.fn();
  const mockProps = {
    newAccount: mockNewAccount,
  };

  let component;
  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <MemoryRouter>
          <SignIn {...mockProps} />
        </MemoryRouter>
      </Provider>
    );
  });

  afterEach(() => {
    cleanup();
  });

  it("should render SignIn", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should contain 'Login' text", () => {
    expect(screen.getByText("Login")).toBeInTheDocument();
  });

  it("should contain email and password inputs", () => {
    expect(screen.getByPlaceholderText("email")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("password")).toBeInTheDocument();
  });

  it("should contain Persistence button", () => {
    expect(screen.getByTestId("signin-persistence")).toBeInTheDocument();
  });

  it("should fire 'setPersistence' when click on switch-button", () => {
    fireEvent.click(screen.getByTestId("signin-persistence"));
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      actionSetPersistence(!mockPersistence)
    );
  });

  it("should contain 'Log in' and 'Sign up' buttons", () => {
    expect(screen.getByText("Log in")).toBeInTheDocument();
    expect(screen.getByText("Sign up")).toBeInTheDocument();
  });

  it("should fire 'handleSubmit' when click on 'Log in' button", () => {
    const email = "email@email.com";
    const password = "12345";

    const emailInput = screen.getByPlaceholderText("email");
    fireEvent.change(emailInput, { target: { value: email } });

    const passwordInput = screen.getByPlaceholderText("password");
    fireEvent.change(passwordInput, { target: { value: password } });

    expect(emailInput).toHaveValue(email);
    expect(passwordInput).toHaveValue(password);

    fireEvent.click(screen.getByText("Log in"));
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      actionEmailSignIn({ email, password })
    );
  });

  it("should fire 'newAccount' when click on 'Sign Up' button", () => {
    fireEvent.click(screen.getByText("Sign up"));
    expect(mockNewAccount).toHaveBeenCalledTimes(1);
  });
});
