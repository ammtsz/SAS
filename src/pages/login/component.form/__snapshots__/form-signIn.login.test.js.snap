// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component login-page <SignIn /> should render SignIn 1`] = `
<DocumentFragment>
  <h4
    class="sc-eCssSg iRCHgB"
  >
    Login
  </h4>
  <form
    class="sc-jSgupP ddDNyR"
  >
    <div
      class="sc-iBPRYJ gYooWe form-input__box"
    >
      <label
        aria-label="email"
        for="signin-email"
      >
        <i
          class="fas fa-envelope"
        />
      </label>
      <input
        class="sc-fubCfw fXNbEQ form-input__input"
        id="signin-email"
        name="email"
        placeholder="email"
        type="email"
      />
    </div>
    <div
      class="sc-iBPRYJ gYooWe form-input__box"
    >
      <label
        aria-label="password"
        for="signin-password"
      >
        <i
          class="fas fa-lock"
        />
      </label>
      <input
        class="sc-fubCfw fXNbEQ form-input__input"
        id="signin-password"
        name="password"
        placeholder="password"
        type="password"
      />
    </div>
    <div
      class="sc-pFZIQ ejEntE"
    >
      <label
        class="sc-jrAGrp eClLHT"
      >
        <input
          class="sc-iqHYGH jpHUlN"
          data-testid="signin-persistence"
          id="button-switch--signin-persistence"
          type="checkbox"
        />
        <span
          class="sc-kEjbxe WmdIn slider round"
        />
      </label>
      <label
        class="button-switch--label"
        for="button-switch--signin-persistence"
      >
        keep me logged in
      </label>
    </div>
    <button
      class="sc-gsTCUz sc-gKsewC dfZqOT kUNQyM"
      type="submit"
    >
      Log in
    </button>
    <button
      class="sc-gsTCUz sc-gKsewC dfZqOT kUNQyM"
    >
      Sign up
    </button>
  </form>
</DocumentFragment>
`;
